db/system_keyspace.cc

```diff
@@ -2642,6 +2654,310 @@ class clients_table : public streaming_virtual_table {
     }
 };
 
+class cluster_describe_table : public streaming_virtual_table {
```

Comment:
I've added description of how I implemented the describe statement (I hope it's detailed enough) and also renamed the virtual tables but I've used `elements_description` instead of `table_description`, since this table contains information about all elements (I know that previous name was totally misleading 😅)

Reply:
What the code in db/system_keyspace.cc does is it basically gets all elements (e.g. keyspaces, tables, udfs ...), gets its descriptions and puts that into virtual table. More of the logic (determining what the describe statement wants to describe, with internal data or not) is in the describe_statement.cc.